static/app/components/narrowLayout.tsx

```diff
@@ -1,58 +1,54 @@
-import {Component} from 'react';
+import {useEffect} from 'react';
 import styled from '@emotion/styled';
 
 import {logout} from 'sentry/actionCreators/account';
-import {Client} from 'sentry/api';
 import {IconSentry} from 'sentry/icons';
 import {t} from 'sentry/locale';
+import useApi from 'sentry/utils/useApi';
 
-type Props = {
+type Props = React.PropsWithChildren<{
   maxWidth?: string;
   showLogout?: boolean;
-};
+}>;
 
-class NarrowLayout extends Component<Props> {
-  UNSAFE_componentWillMount() {
+function NarrowLayout({maxWidth, showLogout, children}: Props) {
+  const api = useApi();
+
+  useEffect(() => {
     document.body.classList.add('narrow');
-  }
 
-  componentWillUnmount() {
-    this.api.clear();
-    document.body.classList.remove('narrow');
+    return () => document.body.classList.remove('narrow');
+  }, []);
+
+  async function handleLogout() {
+    await logout(api);
+    window.location.assign('/auth/login');
   }
 
-  private api = new Client();
-
-  handleLogout = () => {
-    logout(this.api).then(() => window.location.assign('/auth/login'));
-  };
-
-  render() {
-    return (
-      <div className="app">
-        <div className="pattern-bg" />
-        <div className="container" style={{maxWidth: this.props.maxWidth}}>
-          <div className="box box-modal">
-            <div className="box-header">
-              <a href="/">
-                <IconSentry size="lg" />
+  return (
+    <div className="app">
+      <div className="pattern-bg" />
+      <div className="container" style={{maxWidth}}>
+        <div className="box box-modal">
+          <div className="box-header">
+            <a href="/">
```

Comment:
since we are working on this PR. do you think we can replace the `a `element with our `Link` component?

Reply:
We should just fix all the css here, will come back later and do it.